### ISSUE TITLE
web3: gas price strategy

### ISSUE THREAD

**Issue by palango** [raiden-network/raiden#1473]
Currently we get the gas price from the ethereum node by calling the `eth_gasPrice` function.

Web3 offers different gas price strategies. The one we currently use is available as well as a more sophisticated time based version that makes sure that transactions get mined with a certain probability in a given timeframe. (This is similar to what Eth Gas Station provides)
See https://web3py.readthedocs.io/en/stable/gas_price.html#module-web3.gas_strategies.rpc)

When switch to the time based strategy we should define what our goal reading transaction delays are.

**Comment by palango** [comment]
This would be nice to have for Red Eyes. Added it to the milestone, we can remove it if we don't manage to do this.

**Comment by palango** [comment]
Blocked by https://github.com/ethereum/web3.py/issues/979
